### PR TITLE
Insert errors test

### PR DESCRIPTION
**Pull request by imjfckm**
Few minor fixes:
* Make API consistent about using our types in few places

And tests for faulty cache inserts - for now just read/write inserts in non-lazy-write modes, read inserts in lazy writes, write inserts in lazy-write modes will come in next PR

### CODE REVIEW

File: test/functional/api/cas/cache.py [hunk 1]
```diff
@@ -68,6 +68,9 @@ def get_status(self):
         status = self.get_cache_statistics()["status"].replace(' ', '_')
         return CacheStatus[status]
 
+    def get_size(self):
+        return self.get_cache_statistics()["cache size"]
```
Comment (rafalste) [inline test/functional/api/cas/cache.py]: If pull request #218 gets merged, this line should look like this:
`return self.get_cache_statistics().config_stats.cache_size`

Comment (rafalste) [reply test/functional/api/cas/cache.py]: ...or like this:
`return self.get_cache_stats().config_stats.cache_size`
TBD

File: test/functional/api/cas/cache.py [hunk 2]
```diff
@@ -13,8 +13,8 @@
 
 
 class Cache:
-    def __init__(self, device_system_path):
-        self.cache_device = Device(device_system_path)
+    def __init__(self, device):
```
Comment (micrakow) [inline test/functional/api/cas/cache.py]: how about 'Device' type-hint for arg?

File: test/functional/api/cas/cache.py [hunk 3]
```diff
@@ -68,6 +68,9 @@ def get_status(self):
         status = self.get_cache_statistics()["status"].replace(' ', '_')
         return CacheStatus[status]
 
+    def get_size(self):
```
Comment (micrakow) [inline test/functional/api/cas/cache.py]: how about making 'size' @property instead?

File: test/functional/tests/fault/test_cache_insert_error.py [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    TestRun.LOGGER.info("Starting cache")
+    cache = casadm.start_cache(
+        error_device, cache_mode=cache_mode, cache_line_size=cache_line_size, force=True
+    )
+    result = cache.set_seq_cutoff_policy(SeqCutOffPolicy.never)
+    if result.exit_code:
+        TestRun.LOGGER.exception("Couldn't set seq cutoff policy")
+    result = cache.set_cleaning_policy(CleaningPolicy.nop)
+    if result.exit_code:
+        TestRun.LOGGER.exception("Couldn't set seq cleaning policy")
```
Comment (micrakow) [inline test/functional/tests/fault/test_cache_insert_error.py]: 'seq' word not needed here

File: test/functional/api/cas/cache.py (hunk 2 above)
Comment (imjfckm) [reply test/functional/api/cas/cache.py]: Added

File: test/functional/api/cas/cache.py (hunk 3 above)
Comment (imjfckm) [reply test/functional/api/cas/cache.py]: Done!

File: test/functional/tests/fault/test_cache_insert_error.py (hunk 4 above)
Comment (imjfckm) [reply test/functional/tests/fault/test_cache_insert_error.py]: Missed that :)

File: test/functional/api/cas/cache.py [hunk 5]
```diff
@@ -68,6 +68,10 @@ def get_status(self):
         status = self.get_cache_statistics()["status"].replace(' ', '_')
         return CacheStatus[status]
 
+    @property
+    def size(self):
+        return self.get_cache_statistics()["cache size"]
```
Comment (rafalste) [inline test/functional/api/cas/cache.py]: If pull request #218 gets merged this line should look like this:
`return self.get_cache_statistics().config_stats.cache_size`

Comment (imjfckm) [reply test/functional/api/cas/cache.py]: Done

File: test/functional/tests/cli/test_cli_start_stop.py [hunk 6]
```diff
@@ -50,11 +50,12 @@ def test_cli_start_stop_default_value(shortcut):
...
-    cache_device.create_partitions([Size(500, Unit.MebiByte)])
+    cache_device.create_partitions([Size(50, Unit.MebiByte)])
     cache_device = cache_device.partitions[0]
     cache = casadm.start_cache(cache_device, shortcut=shortcut, force=True)
 
     core_device = TestRun.disks['core']
+
     casadm.add_core(cache, core_device, shortcut=shortcut)
 
     caches = casadm_parser.get_caches()
```
[the remark this answers is not in the document]
Comment (imjfckm) [reply test/functional/tests/cli/test_cli_start_stop.py]: Done

File: test/functional/tests/conftest.py [hunk 7]
```diff
@@ -19,6 +19,7 @@
 from api.cas import git
 from test_utils.os_utils import Udev, kill_all_io
 from test_tools.disk_utils import PartitionTable, create_partition_table
+from test_tools.device_mapper import DeviceMapper
```
[the remark this answers is not in the document]
Comment (imjfckm) [reply test/functional/tests/conftest.py]: Done

File: test/functional/tests/fault/test_cache_insert_error.py [hunk 8]
```diff
@@ -0,0 +1,188 @@
...
+          Validate CAS ability to handle write errors while it tries to insert
+          cache lines. For lazy writes cache modes (WO, WB) issue only reads.
+        pass_criteria:
+          - No I/O errors returned to the user
+          - Cache write error statistics are counted properly
+          - No cache line gets inserted into cache
+    """
+    cache, core, core_device = prepare_configuration(cache_mode, cache_line_size)
+
+    fio_cmd = (
```
[the remark this answers is not in the document]
Comment (imjfckm) [reply test/functional/tests/fault/test_cache_insert_error.py]: Done

File: test/functional/tests/fault/test_cache_insert_error.py [hunk 9]
```diff
@@ -0,0 +1,188 @@
...
+        .direct()
+    )
+    if cache_mode in [CacheMode.WB, CacheMode.WO]:
+        fio_cmd = fio_cmd.read_write(ReadWrite.randread)
+    else:
+        fio_cmd = fio_cmd.read_write(ReadWrite.randrw).verify_pattern().verify(VerifyMethod.pattern)
+
+    with TestRun.step("Run fio and verify no errors present"):
+        try:
+            fio_errors = fio_cmd.run()[0].total_errors()
```
[the remark this answers is not in the document]
Comment (imjfckm) [reply test/functional/tests/fault/test_cache_insert_error.py]: Done

**Pull request merged**